gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinClusterConfigurationKeys.java

```diff
@@ -50,6 +50,8 @@ public class GobblinClusterConfigurationKeys {
   public static final boolean DEFAULT_STANDALONE_CLUSTER_MODE = false;
   // Root working directory for Gobblin cluster
   public static final String CLUSTER_WORK_DIR = GOBBLIN_CLUSTER_PREFIX + "workDir";
+  // Root working dir without appending the application name, keeping CLUSTER_WORK_DIR property for backward compatibility
+  public static final String CLUSTER_ABSOLUTE_WORK_DIR = GOBBLIN_CLUSTER_PREFIX + "absolute.workDir";
```

Comment:
two thoughts here:
1. `gobblin.cluster.workDir` is already typically an absolute path.  maybe this would be `gobblin.cluster.exact.workDir`
2. not appending the job/app name seems problematic with the potential for cross-job collisions (e.g. if two different jobs by the same user intend to use the same exact dir).  isn't there a danger there?

Reply:
the expectation here is that the exact workdir will have the job_name appended to it hence no chance of there being collisions, and easier to identify the job associated to it via GaaS

Reply:
not sure if I've missed some critical details here... but who's appending the job_name to this prop?

in general, my pref wouldn't be to push the uniqueness requirement to the caller, but rather to encapsulate within here the choice of a job-unique dir, by taking in a prop for the "base dir" and then auto-appending the particular job name to that

Reply:
It's done by the caller (config file), which can dynamically use properties in the job to append to the folder. It also can append yarn application ID in this config file if you so desire by determining the path at runtime. The issue is that a lot of the other files/folders generated by an E2E Gobblin cluster on yarn app does not also append the yarn application id (leak of abstraction). This approach lets there be more explicit control on the caller side to determine their folder behaviors.